rpcserver.go

```diff
@@ -2476,6 +2477,40 @@ func handleGetNetworkHashPS(s *rpcServer, cmd interface{}, closeChan <-chan stru
 	return hashesPerSec.Int64(), nil
 }
 
+// handleGetNodeAddresses implements the getnodeaddresses command.
+func handleGetNodeAddresses(s *rpcServer, cmd interface{}, closeChan <-chan struct{}) (interface{}, error) {
+	c := cmd.(*btcjson.GetNodeAddressesCmd)
+
+	count := int32(1)
+	if c.Count != nil {
+		count = *c.Count
+		if count <= 0 {
+			return nil, &btcjson.RPCError{
+				Code:    btcjson.ErrRPCInvalidParameter,
+				Message: "Address count out of range",
+			}
+		}
+	}
+
+	nodes := s.cfg.ConnMgr.NodeAddresses()
+	if int32(len(nodes)) < count {
+		count = int32(len(nodes))
+	}
```

Comment:
Will an extra assignment be better than traversing `nodes` twice?

```suggestion
	if numOfNodes := int32(len(nodes)); if  numOfNodes < count {
		count = numOfNodes
	}
```

Reply:
Implementations of `len` should be fast because slice knows its length. I do like the reduced repetition of your suggestion especially since `int32` conversion is duplicated here. I'm keen to name this variable something short like `nn` given its narrow scope.